functions.php

```diff
@@ -45,6 +45,24 @@ function getParamOnType($field_name,$index,$project_id)
     return "[" . $field_name . "] = '" . $index . "'";
 }
 
+function getEthnicityCondition($colTypeMax,$study,$project_id)
+{
+    $condition = "";
+    $type = getFieldType($study, $project_id);
+    for($i=2;$i<$colTypeMax;$i++){
+        if($type == "checkbox"){
+            $condition .= "[" . $study . "(".$i.")] = '1'";
+        }else{
+            $condition .= "[" . $study . "] = '".$i."'";
+        }
+
+        if($i != ($colTypeMax-1)){
+            $condition .= " OR ";
+        }
```

Comment:
@knil-maloon I was doing something like this in my project. In the future, I think we should put our heads together and write some abstract methods in the module framework that automatically creates filter logic from arrays,,,,

Reply:
I agree. That could be a new function. There's also a tinker I did for yes/no & true/false that could be updated to getChoiceLabels function but for this particular module I had to add it instead of updating the function because some institutions have older REDCap versions. It's one thing on my to do list when I get the chance ;)